### ISSUE TITLE
How to read errors in the Sendgrid response object.

### ISSUE THREAD

**Issue by pradeep864** [sendgrid/sendgrid-csharp#663]
#### Issue Summary

I am using **Sendgrid nuget package v9.90** to send emails. In the event of any bad requests, i expect to see an error object in the response. I tried sending emails using below methods but still couldn't find the errors section in the response object

1. MailHelper.CreateSingleEmail
2. client.RequestAsync(SendGridClient.Method.POST, json.ToString(), urlPath: "mail/send")

Could you please help me understanding the way to access the errors in the response object. 

#### Steps to Reproduce

Below is my code

        public static async Task<HttpStatusCode> SendGridasync(MailMessage message)
        {
            var apiKey = APIKey;
            var client = new SendGridClient(apiKey);

            string data = @"{
              'personalizations': [
                {
                  'to': [
                    {
                      'email': 'test@test.com'
                    }
                  ],
                  'subject': 'Hello World from the SendGrid C# Library!'
                }
              ],
              'from': {
                'email': 'test@example.com'
              }
            }";

            var json = JsonConvert.DeserializeObject<Object>(data);
            var response = await client.RequestAsync(SendGridClient.Method.POST, json.ToString(), urlPath: "mail/send");
            return response.StatusCode;

            //,
            //  'content': [
            //    {
            //      'type': 'text/plain',
            //      'value': 'Hello, Email!'
            //    }
            //  ]
        }

#### Technical details:
* Platform ( ASP.NET)
* .NET Version: 4.6.2

**Comment by thinkingserious** [comment]
Hi @pradeep864,

[Here](https://github.com/sendgrid/sendgrid-csharp/blob/master/TROUBLESHOOTING.md#error) you go. I hope that helps!

With Best Regards,

Elmer

**Comment by pradeep864** [comment]
Hi @thinkingserious,

I appreciate your help on this.

Regards,
Pradeeep

**Comment by LCvijovic09** [comment]
Hi, I would like to report that this method did not work for me. I am on version 9.7.0. I keep getting an error while using ReadAsStringAsync(). The error says something like this: Can't find method get_Body() in SendGrid (please note that this is not exectly what it says, I can't remeber how exact text was). I am using WebApi and seems to be repated with HttpContent (SendGrid responses body).
Thank you and best regards.

**Comment by thinkingserious** [comment]
Hello @LCvijovic09,

In order to best assist, could you please provide the specific error and/or stack trace? Also, the source code could be helpful for reproducing the issue.

With Best Regards,

Elmer

**Comment by LCvijovic09** [comment]
Hi Elmer,
unfortenatlly I can't provide it. I figured the issue after several try/fails. The issue was said "System.[something].HttpContent SendGrid.HttpContent.get_Body() is not a method [or could not find a method]". I know this is now preatty hard to follow through, as I can't provide much details at all, but to me this seems to be related with using of IHttpResponse ibstead of HttpResponse for ApiMethods. I might be wrong, but still if you find some time maybe you could investigate a little, if somebody has the same issue, or provide a little explanation on this issue.
Thank you. Best regards..

**Comment by thinkingserious** [comment]
Hi @LCvijovic09,

Have you tried upgrading to the latest version of this SDK? (v9.11.0)

Perhaps you can send the relevant code to us via email? dx@sendgrid.com.

Thanks!

**Comment by stiliqn-angelov** [comment]
Hi @tappdarden,
I am encountering the same error while using v9.11.0

`[MissingMethodException: Method not found: 'System.Net.Http.HttpContent SendGrid.Response.get_Body()'.]`

**Comment by thinkingserious** [comment]
Hello @stiliqn-angelov,

Could you please provide the source code that is producing this error? If you would rather not post it here, please send it to us via email so that we may try to reproduce the error: dx@sendgrid.com

Thanks!

With Best Regards,

Elmer

**Comment by niranjankrsingh** [comment]
Hi @tappdarden @thinkingserious 
I am also facing same issue on production server. I have sent you detail in email dx@sendgrid.com.

Method not found: 'System.Net.Http.HttpContent SendGrid.Response.get_Body()'

Thanks
Niranjan

**Comment by LCvijovic09** [comment]
Ok, so I made it work somehow. Here is the solution, first, put a breakpoint after you get the response. In "immidiate window" in VisualStudio type: namOfResponseVariable.Body.ReadAsStringAsync().Result
Thats it. :) Best of luck.

**Comment by LCvijovic09** [comment]
Also, one more thing. This may very well vary on the error type. For example, I had this error when I passed to Bcc or Cc empty list. The error was that I had to pass null, or list that is populated.

**Comment by zeckdude** [comment]
This might be helpful: https://github.com/sendgrid/sendgrid-nodejs/blob/master/use-cases/success-failure-errors.md

**Comment by christophborndigital** [comment]
We also got the error
Method not found: 'System.Net.Http.HttpContent SendGrid.Response.get_Body()'
In our case the SendGrid package was used in a class library, but the web project referencing it had an older version of System.Net.Http installed. Consolidating the versions of System.Net.Http allowed us to get the response body to see the error message.

**Comment by startechsoftwares** [comment]
Hi @thinkingserious ,

I have got into same problem, i am shooting out around (50-100) emails at a time, i want to do some post-processing on results of email status, for example, i have a field in my database called HasMailSent, i want to mark it true if it has went through or false if it hasn't. However after i read the response from the var response = await client.SendEmailAsync(msg) i gets Accepted, but how to ensure which emails has went through, or it could be that this response comes if all the emails are accepted, but my concern is i want the response about individual emails.

Please help me on this.

Thanks,
Abbas

**Comment by childish-sambino** [comment]
@startechsoftwares Relevant comment here: https://github.com/sendgrid/sendgrid-php/issues/202#issuecomment-189687070

**Comment by worthy7** [comment]
> Hi @pradeep864,
> 
> [Here](https://github.com/sendgrid/sendgrid-csharp/blob/master/TROUBLESHOOTING.md#error) you go. I hope that helps!
> 
> With Best Regards,
> 
> Elmer

This doesn't seem to work, there is no 'HttpErrorAsException' option

**Comment by worthy7** [comment]
Never mind, the package literally just updated and I grabbed the new version thanks.
